### DISCUSSION TITLE
Is there a way to let raw attributes like `class` or `id` to be automatically applied to root node of component ?

### DISCUSSION THREAD

**Discussion by leo91000** [leptos-rs/leptos#399] [Q&A] [answered]
Is there a way to pass attributes like "class" or "id" to the root node ?

If I do this
```rust
#[component]
fn MyComponent(cx: Scope) -> impl IntoView {
    view! { cx
         <div>Foo</div>
    }
}

#[component]
fn App(cx: Scope) -> impl IntoView {
   view! { cx,
        <MyComponent class="foobar" />
   }
}
```

I have an error saying that `MyComponent` doesn't have a class props. 
Is there a way to pass down raw attributes like `class` or `id` ?
Do I have to redeclare each time the props ?

**Comment by gbj** [comment, marked as answer]
The short answer: No, you can't pass HTML element attributes directly to components are component props. They're conceptually different things: think of the component props as function arguments. The compiler is helpfully telling you that there's no `class` prop on `MyComponent`, which is true. You can add one and pass it through to the HTML element.

```rust
#[component]
fn MyComponent<'a>(cx: Scope, class: &'a str) -> impl IntoView {
    view! { cx
         <div class=class>Foo</div>
    }
}
```
or use a String or whatever type you'd like, if you don't want to deal with the lifetime annotation
